airbyte-webapp/src/packages/cloud/components/experiments/SignupSourceDropdown/utils.tsx

```diff
@@ -0,0 +1,43 @@
+import { ConnectorIcon } from "components/ConnectorIcon";
+
+import { Connector } from "core/domain/connector";
+import { ReleaseStage, SourceDefinitionRead } from "core/request/AirbyteClient";
+import { naturalComparator } from "utils/objects";
+
+/**
+ * Returns the order for a specific release stage label. This will define
+ * in what order the different release stages are shown inside the select.
+ * They will be shown in an increasing order (i.e. 0 on top)
+ */
+const getOrderForReleaseStage = (stage?: ReleaseStage): number => {
+  switch (stage) {
+    case ReleaseStage.beta:
+      return 1;
+    case ReleaseStage.alpha:
+      return 2;
+    default:
+      return 0;
+  }
+};
+interface ServiceDropdownOption {
+  label: string;
+  value: string;
+  img: JSX.Element;
+  releaseStage: ReleaseStage | undefined;
+}
+const transformConnectorDefinitionToDropdownOption = (item: SourceDefinitionRead): ServiceDropdownOption => ({
+  label: item.name,
+  value: Connector.id(item),
+  img: <ConnectorIcon icon={item.icon} />,
+  releaseStage: item.releaseStage,
+});
+
+const sortByReleaseStage = (a: ServiceDropdownOption, b: ServiceDropdownOption) => {
+  if (a.releaseStage !== b.releaseStage) {
+    return getOrderForReleaseStage(a.releaseStage) - getOrderForReleaseStage(b.releaseStage);
+  }
+  return naturalComparator(a.label, b.label);
+};
+
+export const getSortedDropdownData = (availableConnectorDefinitions: SourceDefinitionRead[]): ServiceDropdownOption[] =>
```

Comment:
We only sort by release stage, but if we want to replicate `connector.override` here in the future, we can absolutely do it

Reply:
I think that's fine for now. If we'd keep this feature, we'd anyway need to remove this second component again, so I don't think we need to have the full functionality for now.